text/204-golang-bindings.md

```diff
@@ -0,0 +1,655 @@
+  }
+}
+
+```
+
+#### Case 2: Extending Typescript datatype interfaces
+
+For Typescript datatype interfaces that extend another datatype interface, the corresponding interfaces and implementing structs would be generated in
+go, with the extending interface embedding the extended one, e.g.:
+
+```ts
+export interface BaseServiceOptions {
+  readonly cluster: ICluster;
+  readonly desiredCount?: number;
+  readonly serviceName?: string;
+  readonly maxHealthyPercent?: number;
+  readonly minHealthyPercent?: number;
+  readonly healthCheckGracePeriod?: Duration;
+  readonly cloudMapOptions?: CloudMapOptions;
+  readonly propagateTags?: PropagatedTagSource;
+  readonly enableECSManagedTags?: boolean;
+  readonly deploymentController?: DeploymentController;
+}
+
+export interface BaseServiceProps extends BaseServiceOptions {
+  readonly launchType: LaunchType;
+}
+```
+
+As before, the Go translation would have the getter methods (e.g. readonly cluster) defined in a the generated interface, which correspond to
+non-exported struct members in the struct that implements each getter.
+
+```go
+package ecs
+
+type BaseServiceOptionsIface interface {
+    Cluster()                ICluster
+    DesiredCount()           *int
+    ServiceName()            *string
+    MaxHealthyPercent()      *int
+    MinHealthyPercent()      *int
+    HealthCheckGracePeriod() *Duration
+    CloudMapOptions()        CloudMapOptions
+    PropagateTags()          PropagatedTagSource
+    EnableECSManagedTags()   *bool
+    DeploymentController()   DeploymentController
+}
+
+type BaseServicePropsIface interface {
+    serviceOptions           BaseServiceOptionsIface  // embeddeded interface
+    LaunchType()             LaunchType
+}
+
+type BaseServiceOptions struct {
+    cluster                ICluster
+    desiredCount           *int
+    serviceName            *string
+    maxHealthyPercent      *int
+    minHealthyPercent      *int
+    healthCheckGracePeriod *Duration
+    cloudMapOptions        CloudMapOptions
+    propagateTags          PropagatedTagSource
+    enableECSManagedTags   *bool
+    deploymentController   DeploymentController
+}
+
+type BaseServiceProps struct {
+    baseOptions         BaseServiceOptions  // embedded (anonymous?) field
+    launchType          LaunchType
+}
+
+func (o BaseServiceOptions) Cluster() ICluster { return o.cluster }
+func (o BaseServiceOptions) DesiredCount() int{ return o.desiredCount }
+
+// ... etc
+
+```
+
+This allows the embedding interface (here, `BaseServicePropsIFace`) to “inherit” all the methods defined in the embedded interface
+(`BaseServiceOptionsIFace`). Similarly, the embedding struct (`BaseServiceProps`) would have access to all the properties of the embedded struct
+(`BaseServiceOptions`) through having it as an anonymous field (See: [Go playground example](https://play.golang.org/p/XRpTH3LN6bx)).
+
+```go
+props := BaseServiceProps{
+  BaseServiceOptions{
+    ServiceName: "myService",
+    ...
+  },
```

Comment:
If this is the usage, I don't think it's good enough because it requires that users intimately understand the hierarchy  between the data types.

Is it possible to find a solution where from a user's perspective this is flat?

Reply:
How is this different from how extended interfaces are used in Typescript?

Reply:
No, the IDE will flatten the namespace so it all appears as a flat list.